// nestjs-mailgun@4.0.3/dist/mailgun.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var MailgunModule_1;
Object.defineProperty(exports, "__esModule", { value: true });
exports.MailgunModule = void 0;
const common_1 = require("@nestjs/common");
const constants_1 = require("./constants");
const mailgun_service_1 = require("./mailgun.service");
let MailgunModule = exports.MailgunModule = MailgunModule_1 = class MailgunModule {
    static forRoot(config) {
        return {
            module: MailgunModule_1,
            providers: [
                { provide: constants_1.MAILGUN_CONFIGURATION, useValue: config },
                mailgun_service_1.MailgunService,
            ],
            exports: [mailgun_service_1.MailgunService],
        };
    }
    static forAsyncRoot(config) {
        return {
            module: MailgunModule_1,
            imports: config.imports || [],
            providers: [this.createAsyncProviders(config), mailgun_service_1.MailgunService],
            exports: [mailgun_service_1.MailgunService],
        };
    }
    static createAsyncProviders(options) {
        return {
            provide: constants_1.MAILGUN_CONFIGURATION,
            useFactory: options.useFactory,
            inject: options.inject || [],
        };
    }
};
exports.MailgunModule = MailgunModule = MailgunModule_1 = __decorate([
    (0, common_1.Module)({})
], MailgunModule);
